inst/skeleton/rcppsmc_hello_world.cpp

```diff
@@ -0,0 +1,23 @@
+// -*- mode: C++; c-indent-level: 4; c-basic-offset: 4; indent-tabs-mode: nil; -*-
+
+// we only include RcppSMC.h which pulls Rcpp.h in for us
+#include "RcppSMC.h"
+
+// via the depends attribute we tell Rcpp to create hooks for
+// RcppSMC so that the build process will know what to do
+//
+// [[Rcpp::depends(RcppSMC)]]
+
+// simple example of computing stable calculations of the logarithm of the sum
+// of weights which is of interest for computing e.g. normalising constants
+//
+// via the exports attribute we tell Rcpp to make this function
+// available from R
+//
+// [[Rcpp::export]]
+Rcpp::List rcppsmc_logNormWeightsCpp(arma::vec logWeights){
+  double logSum = smc::stableLogSumWeights(logWeights);
+  logWeights -= logSum;
+  return Rcpp::List::create(Rcpp::Named("logSum") = logSum,
+                            Rcpp::Named("logNormWeights") = logWeights);
```

Comment:
If the results are both scalars it it easier to return a numeric vector of length 2?

Reply:
@eddelbuettel that's tricky. Essentially this is an example from the JSS paper, which is likely to change anyway. I wanted to have consistency between what the skeleton does ("exact" line by line) and what the JSS will provide as an example, mostly for clarity purposes.

SInce this is likely to change anyway, as we might update the example from the paper, I can of course change it __*now*__  but it it might change in the future anyway. What do you think?